srml/contract/src/lib.rs

```diff
@@ -309,6 +318,14 @@ impl<T: Trait> Module<T> {
 
 			// Then deposit all events produced.
 			ctx.events.into_iter().for_each(Self::deposit_event);
+
+			// Perform suicides.
+			//
+			// NOTE: this should go after the commit to the storage, since the storage changes
+			// can overwrite these transfers.
+			ctx.suicides.into_iter().for_each(|note| {
+				account_db::commit_suicide::<T>(&note.who, &note.inherent);
```

Comment:
What will happen if we suicide `A -> B` and then `B -> A`; what if we do that in reverse order?
I'm not sure what the right solution should be, I guess it's fine if suicides happen in the same order as "scheduled" and also if a sucided account can be "revived" by another suicide. Let's just make sure it works correctly.

Reply:
In this scheme a contract can't be revived, since all contracts in suicide list will be removed (by `purge_account`) due to draining funds or manually (for the case `existential_deposit = 0`). But yeah I think it's possible to send funds to a dead contract.